wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -0,0 +1,279 @@
+      case BIP44ChainType.External =>
+        addressDAO.findMostRecentExternal(accountIndex)
+      case BIP44ChainType.Change =>
+        addressDAO.findMostRecentChange(accountIndex)
+    }
+
+    lastAddrOptF.flatMap { lastAddrOpt =>
+      val addrPath = lastAddrOpt match {
+        case Some(addr) => addr.path.next.address
+        case None =>
+          val account = BIP44Account(bip44Coin, accountIndex)
+          val chain = account.toChain(chainType)
+          BIP44Address(chain, 0)
+      }
+
+      val addressDb =
+        AddressDbHelper
+          .getP2WPKHAddress(xpriv, addrPath.toPath, networkParameters)
+      addressDAO.create(addressDb).map(_.address)
+    }
+  }
+
+  /**
+    * right now only generates P2WPKH addresses
+    *
+    * @inheritdoc
+    */
+  override def getNewAddress(accountIndex: Int = 0): Future[BitcoinAddress] = {
+    getNewAddressHelper(accountIndex, BIP44ChainType.External)
+  }
+
+  /** Generates a new change address */
+  private def getNewChangeAddress(
+      accountIndex: Int = 0): Future[BitcoinAddress] = {
+    getNewAddressHelper(accountIndex, BIP44ChainType.Change)
+  }
+
+  override def sendToAddress(
+      address: BitcoinAddress,
+      amount: CurrencyUnit,
+      accountIndex: Int = 0): Future[Transaction] = {
+    for {
+      change <- getNewChangeAddress(accountIndex)
+      fee <- feeProvider.getFee(1)
+      walletUtxos <- listUtxos()
+      txBuilder <- {
+        val destinations: Seq[TransactionOutput] = List(
+          TransactionOutput(amount, address.scriptPubKey))
+
+        // currencly just grabs one utxos, throws if can't find big enough
+        // todo: implement coin selection
+        val utxos: List[BitcoinUTXOSpendingInfo] =
+          List(walletUtxos.find(_.value >= amount).get.toUTXOSpendingInfo(this))
+
+        BitcoinTxBuilder(destinations,
+                         utxos,
+                         fee,
+                         changeSPK = change.scriptPubKey,
+                         networkParameters)
+      }
+      signed <- txBuilder.sign
+      /* todo: add change output to UTXO DB
+       _ <- {
+        val changeVout = ???
+        addUtxo(signed, changeVout)
+      } */
+    } yield {
+      signed
+    }
+  }
+
+  /**
+    * @inheritdoc
+    */
+  override def unlock(passphrase: String): Future[Try[UnlockedWalletApi]] = ???
+
+  /**
+    * @inheritdoc
+    */
+  // override def getAccounts: Future[Vector[BIP44Account]] = ???
+
+  /**
+    * @inheritdoc
+    */
+  // override def createNewAccount: Future[Try[WalletApi]] = ???
+
+}
+
+// todo: create multiple wallets, need to maintain multiple databases
+object Wallet extends CreateWalletApi {
+  private case class WalletImpl(
+      mnemonicCode: MnemonicCode,
+      passphrase: String,
+      dbConfig: DbConfig,
+      chainParams: ChainParams)(implicit executionContext: ExecutionContext)
+      extends Wallet {
+    override implicit def ec: ExecutionContext = executionContext
```

Comment:
any reason we don't just do 

```scala
 private case class WalletImpl(
      mnemonicCode: MnemonicCode,
      passphrase: String,
      dbConfig: DbConfig,
      chainParams: ChainParams)(override implicit val ec: ExecutionContext)
      extends Wallet { ... } 
```

Reply:
We cannot have the `override` modifier. That turns `WalletImpl` into `ExecutionContext => Wallet.WalletImpl`. But I agree it's cleaner to give up the override word to save some lines